lib/Runtime/Base/FunctionBody.cpp

```diff
@@ -1912,6 +1912,9 @@ namespace Js
 
         if (fParsed == TRUE)
         {
+            // Restore if the function has nameIdentifier reference, as that name on the left side will not be parsed again while deferparse.
+            funcBody->SetIsNameIdentifierRef(this->GetIsNameIdentifierRef());
```

Comment:
I need to set that as the parser above will change the state. imagine the code is 
`a.b = function() { /*blah*/}`
however in the defer parse we see the code as 
        ` function() { /*blah*/ } `
So I am restoring the previous state.
If you are talking about renaming, I am not sure if this is needed.